File: backend/src/models/katalog_model.js
import database from '../configs/database.js';

class KatalogModel {
    static async getAllKatalog() {
        const [rows] = await database.execute('SELECT * FROM Katalog');
        console.log("KatalogModel.getAllKatalog: ", rows);
        return rows;
    }

    static async getKatalogById(katalog_id) {
        const [rows] = await database.execute('SELECT * FROM Katalog WHERE katalog_id = ?', [katalog_id]);
        console.log("KatalogModel.getKatalogById: ", rows);
        return rows[0];
    }

    static async createKatalog(nama, deskripsi, kategori_id, harga, kondisi, status, nomor_polisi) {
        const [result] = await database.execute('INSERT INTO Katalog (nama, deskripsi, kategori_id, harga, kondisi, status, nomor_polisi) VALUES (?, ?, ?, ?, ?, ?, ?)', [nama, deskripsi, kategori_id, harga, kondisi, status, nomor_polisi]);
        console.log("KatalogModel.createKatalog: ", result);
        return result.insertId;
    }

    static async updateKatalog(katalog_id, nama, deskripsi, kategori_id, harga, kondisi, status, nomor_polisi) {
        const [result] = await database.execute('UPDATE Katalog SET nama = ?, deskripsi = ?, kategori_id = ?, harga = ?, kondisi = ?, status = ?, nomor_polisi = ? WHERE katalog_id = ?', [nama, deskripsi, kategori_id, harga, kondisi, status, nomor_polisi, katalog_id]);
        console.log("KatalogModel.updateKatalog: ", result);
        return result.affectedRows > 0;
    }

    static async deleteKatalog(katalog_id) {
        const [result] = await database.execute('DELETE FROM Katalog WHERE katalog_id = ?', [katalog_id]);
        console.log("KatalogModel.deleteKatalog: ", result);
        return result.affectedRows > 0;
    }

    static async searchKatalog(keyword) {
        const [rows] = await database.execute('SELECT * FROM Katalog WHERE nama LIKE ? OR nomor_polisi LIKE ?', [`%${keyword}%`, `%${keyword}%`]);
        console.log("KatalogModel.searchKatalog: ", rows);
        return rows;
    }

    static async getKatalogByNomorPolisi(nomor_polisi) {
        const [rows] = await database.execute('SELECT * FROM Katalog WHERE nomor_polisi = ?', [nomor_polisi]);
        console.log("KatalogModel.getKatalogByNomorPolisi: ", rows);
        return rows[0];
    }
}

export default KatalogModel;

File: drive_ease/lib/data/models/Katalog_model.dart
import 'package:equatable/equatable.dart';

class KatalogModel extends Equatable {
  final int katalog_id;
  final String nama;
  final String deskripsi;
  final int kategori_id;
  final double harga;
  final String kondisi;
  final String status;
  final String nomor_polisi;
  final String updated_at;

  const KatalogModel({
    required this.katalog_id,
    required this.nama,
    required this.deskripsi,
    required this.kategori_id,
    required this.harga,
    required this.kondisi,
    required this.status,
    required this.nomor_polisi,
    required this.updated_at,
  });

  factory KatalogModel.fromJson(Map<String, dynamic> json) {
    return KatalogModel(
      katalog_id: json['katalog_id'] ?? '',
      nama: json['nama'] ?? '',
      deskripsi: json['deskripsi'] ?? '',
      kategori_id: json['kategori_id'] ?? '',
      harga: (json['harga'] != null) ? double.tryParse(json['harga'].toString()) ?? 0.0 : 0.0,
      kondisi: json['kondisi'] ?? '',
      status: json['status'] ?? '',
      nomor_polisi: json['nomor_polisi'] ?? '',
      updated_at: json['updated_at'] ?? '',
    );
  }

  @override
  List<Object?> get props => [katalog_id, nama, deskripsi, kategori_id, harga, kondisi, status, nomor_polisi, updated_at];
}